gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/FeatureConverter.java

```diff
@@ -75,18 +76,19 @@ private static PolygonFeature toFeatureFromGeometry(
       throw new DataStoreException("Unknown geometry type in feature " + doc.getId() + ": " + type);
     }
     Object coordinates = geometry.get(GEOMETRY_COORDINATES);
-    if (coordinates == null || !coordinates.getClass().isArray()) {
+    if (coordinates == null || !(coordinates instanceof ArrayList)) {
       throw new DataStoreException(
           "Invalid coordinates in feature " + doc.getId() + ": " + coordinates);
     }
     ImmutableList.Builder<Point> vertices = ImmutableList.builder();
-    for (Object point : (Object[]) coordinates) {
-      if (point instanceof GeoPoint) {
-        vertices.add(Point.newBuilder().setLongitude(((GeoPoint) point).getLongitude()).setLatitude(
-            ((GeoPoint) point).getLatitude()).build());
-      } else {
+    for (Object point : (ArrayList<?>) coordinates) {
```

Comment:
Same here; cast to `List` here instead unless we specifically need `ArrayList`.

Reply:
changed ArrayList to List